db/view/view.cc

```diff
@@ -318,6 +320,50 @@ static bool is_partition_key_empty(
     }
 }
 
+// Checks if the result matches the provided view filter
+class view_filter_checking_visitor {
+    const schema& _base;
+    const view_info& _view;
+    const std::vector<bytes>& _pk;
+    const std::vector<bytes>& _ck;
+
+    bool _matches_view_filter = true;
+public:
+    view_filter_checking_visitor(const schema& base, const view_info& view, const std::vector<bytes>& pk, const std::vector<bytes>& ck)
+        : _base(base)
+        , _view(view)
+        , _pk(pk)
+        , _ck(ck)
+    {}
+
+    void accept_new_partition(const partition_key& key, uint64_t row_count) {}
+    void accept_new_partition(uint64_t row_count) {}
+
+    void accept_new_row(const clustering_key& key, const query::result_row_view& static_row, const query::result_row_view& row) {
+        _matches_view_filter &= check_if_matches(static_row, row);
+    }
+
+    void accept_new_row(const query::result_row_view& static_row, const query::result_row_view& row) {
+        _matches_view_filter &= check_if_matches(static_row, row);
+    }
+    void accept_partition_end(const query::result_row_view& static_row) {}
+
+    bool check_if_matches(const query::result_row_view& static_row, const query::result_row_view& row) const {
+        auto selection = cql3::selection::selection::wildcard(_base.shared_from_this());
```

Comment:
Can this be moved to the constructor? It looks independent of static_row/row.

I'm guessing you re-create the visitor for each row, but perhaps that can be optimized.

Reply:
It's probably negligible compared to all the other waste (exploding stuff, using bytes instead of managed_bytes_view), so we can defer it.

Reply:
It is negligible, but it's also a good idea, so I'll apply it - `wildcard()` actually creates a shared pointer, so it's better to make a useless allocation once, for future sake.